File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] score) {
        ArrayList<Integer> tmp = new ArrayList<>();
        int[] answer = {};
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int N = score.length;
        
        for (int i = 0; i < N; i++) {
            pq.offer(score[i]);
            if (pq.size() > k) {
                pq.poll();
            }
            tmp.add(pq.peek());
        }
        answer = new int[tmp.size()];
        for (int i = 0; i < tmp.size(); i++) {
            answer[i] = tmp.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
from heapq import heappush, heappop
def solution(k, score):
    answer = []
    q = []
    N = len(score)
    for i in range(N):
        heappush(q, score[i])
        if len(q) > k:
            heappop(q)
        answer.append(q[0])
    return answer